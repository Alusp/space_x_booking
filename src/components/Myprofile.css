.our-profile {
  display: flex;
  margin-top: 10px;
  align-items: center;
  justify-content: space-around;
}

.our-profile div {
  width: 50%;
  margin: 5%;
}
