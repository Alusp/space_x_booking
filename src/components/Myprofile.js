import { useSelector } from 'react-redux';
import './Myprofile.css';

const Myprofile = () => {
  const allmission = useSelector((state) => state.mission);
  const allRockets = useSelector((state) => state.rockets);

  const rocket = Object.keys(allRockets).map((key) => ({
    reserved: allRockets[key].reserved,
    rocket_name: allRockets[key].rocket_name,
  }));

  const mission = Object.keys(allmission).map((key) => ({
    join: allmission[key].join,
    mission_name: allmission[key].mission_name,
  }));
  return (
    <div className="our-profile">
      <div>
        <table className="mission_table">
          <tr>
            <th>My Missions</th>
          </tr>
          {mission.map((mis) => mis.join && (
          <tr key={mis.mission_id}>
            <td>{mis.mission_name}</td>
          </tr>
          ))}
        </table>
      </div>
      <div>
        <table className="rocket_table">
          <tr>
            <th>My Rockets</th>
          </tr>
          {rocket.map((rock) => rock.reserved && (
          <tr key={rock.rocket_id}>
            <td>{rock.rocket_name}</td>
          </tr>
          ))}
        </table>
      </div>
    </div>
  );
};

export default Myprofile;
