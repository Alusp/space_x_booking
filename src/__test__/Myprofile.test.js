import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Myprofile from '../components/Myprofile';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Myprofile />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
