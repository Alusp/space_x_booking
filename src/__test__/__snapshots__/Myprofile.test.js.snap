// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="our-profile"
>
  <div>
    <table
      className="mission_table"
    >
      <tr>
        <th>
          My Missions
        </th>
      </tr>
    </table>
  </div>
  <div>
    <table
      className="rocket_table"
    >
      <tr>
        <th>
          My Rockets
        </th>
      </tr>
    </table>
  </div>
</div>
`;
